plugin-build/plugin/src/main/java/com/ncorti/ktfmt/gradle/tasks/worker/KtfmtFormatResult.kt

```diff
@@ -0,0 +1,41 @@
+package com.ncorti.ktfmt.gradle.tasks.worker
+
+import java.io.File
+
+internal sealed class KtfmtFormatResult(open val input: File) {
+
+    abstract fun serialize(): String
+
+    data class KtfmtFormatSuccess(override val input: File, val wasCorrectlyFormatted: Boolean) :
+        KtfmtFormatResult(input) {
+
+        override fun serialize(): String = "success,$wasCorrectlyFormatted,${input.path}"
+    }
+
+    data class KtfmtFormatFailure(override val input: File) : KtfmtFormatResult(input) {
+
+        override fun serialize(): String = "failure,false,${input.path}"
+    }
+
+    data class KtfmtFormatSkipped(override val input: File) : KtfmtFormatResult(input) {
+
+        override fun serialize(): String = "skipped,false,${input.path}"
+    }
+
+    companion object {
+        fun parse(str: String): KtfmtFormatResult {
+            val pieces = str.split(",")
+
+            @Suppress("MagicNumber")
+            check(pieces.size == 3) { "Expected three components; got $str" }
+            return when (pieces[0]) {
```

Comment:
```suggestion
            check(pieces.size == 3) { "Expected three components; got $str" }
            val (status, input, wasCorrectlyFormatted) = pieces
            return when (status) {
```

+ update the occurrencies also inside the when cases